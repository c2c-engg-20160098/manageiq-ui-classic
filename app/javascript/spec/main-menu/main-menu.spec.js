import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import MainMenu from '../../components/main-menu/main-menu';
import TopLevel from '../../components/main-menu/top-level';
import SecondLevel from '../../components/main-menu/second-level';
import ThirdLevel from '../../components/main-menu/third-level';

describe('Main menu test', () => {
  let mockNavItems;
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore({
      menuReducer: {
        isVerticalMenuCollapsed: false,
      },
    });
    mockNavItems = [{
      id: 'vi',
      title: 'Cloud Intel',
      iconClass: 'fa fa-dashboard',
      href: '/dashboard/maintab/?tab=vi',
      visible: true,
      active: true,
      type: 'default',
      items: [{
        id: 'dashboard',
        title: 'Dashboard',
        iconClass: null,
        href: '/dashboard/show',
        visible: true,
        active: true,
        type: 'modal',
        items: [],
      }, {
        id: 'chargeback',
        title: 'Chargeback',
        iconClass: null,
        href: '/chargeback/explorer',
        visible: true,
        active: false,
        type: 'big_iframe',
        items: [],
      }],
    }, {
      id: 'compute',
      title: 'Compute',
      iconClass: 'pficon pficon-cpu',
      href: '/dashboard/maintab/?tab=compute',
      visible: true,
      active: false,
      type: 'new_window',
      items: [{
        id: 'clo',
        title: 'Clouds',
        iconClass: 'fa fa-plus',
        href: '/dashboard/maintab/?tab=clo',
        visible: true,
        active: false,
        type: 'new_window',
        items: [{
          id: 'ems_cloud',
          title: 'Providers',
          iconClass: null,
          href: '/ems_cloud/show_list',
          visible: true,
          active: true,
          type: 'new_window',
          items: [],
        }, {
          id: 'availability_zone',
          title: 'Availability Zones',
          iconClass: null,
          href: '/availability_zone/show_list',
          visible: true,
          active: false,
          type: 'default',
          items: [],
        }, {
          id: 'host_aggregate',
          title: 'Host Aggregates',
          iconClass: null,
          href: '/host_aggregate/show_list',
          visible: false,
          active: false,
          type: 'new_window',
          items: [],
        }],
      }],
    }];
  });

  it('should render correctly', () => {
    const wrapper = mount(<Provider store={store}><MainMenu menu={mockNavItems} /></Provider>).find(MainMenu);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render proper level components', () => {
    const wrapper = mount(<Provider store={store}><MainMenu menu={mockNavItems} /></Provider>);
    expect(wrapper.find(TopLevel)).toHaveLength(2);
    expect(wrapper.find(SecondLevel)).toHaveLength(3);
    expect(wrapper.find(ThirdLevel)).toHaveLength(3);
  });

  it('should render active third level components properly', () => {
    const wrapper = mount(<Provider store={store}><MainMenu menu={mockNavItems} /></Provider>);
    expect(wrapper.find(ThirdLevel).find('li.menu-list-group-item.active'))
      .toHaveLength(1);
  });

  it('should render not collapsed vertical navbar with proper classNames', () => {
    const wrapper = mount(<Provider store={store}><MainMenu menu={mockNavItems} /></Provider>);
    expect(wrapper.find('#main-menu').instance().className)
      .toEqual('nav-pf-vertical nav-pf-vertical-with-sub-menus nav-pf-vertical-collapsible-menus');
  });

  it('should render collapsed vertical navbar with proper classNames', () => {
    const collapsedStore = mockStore({
      menuReducer: {
        isVerticalMenuCollapsed: true,
      },
    });
    const wrapper = mount(<Provider store={collapsedStore}><MainMenu menu={mockNavItems} /></Provider>);
    expect(wrapper.find('#main-menu').instance().className)
      .toEqual('nav-pf-vertical nav-pf-vertical-with-sub-menus nav-pf-vertical-collapsible-menus collapsed');
  });
});
