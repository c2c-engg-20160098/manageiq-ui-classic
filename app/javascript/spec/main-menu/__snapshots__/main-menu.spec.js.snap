// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main menu test should render correctly 1`] = `
<MainMenu
  menu={
    Array [
      Object {
        "active": true,
        "href": "/dashboard/maintab/?tab=vi",
        "iconClass": "fa fa-dashboard",
        "id": "vi",
        "items": Array [
          Object {
            "active": true,
            "href": "/dashboard/show",
            "iconClass": null,
            "id": "dashboard",
            "items": Array [],
            "title": "Dashboard",
            "type": "modal",
            "visible": true,
          },
          Object {
            "active": false,
            "href": "/chargeback/explorer",
            "iconClass": null,
            "id": "chargeback",
            "items": Array [],
            "title": "Chargeback",
            "type": "big_iframe",
            "visible": true,
          },
        ],
        "title": "Cloud Intel",
        "type": "default",
        "visible": true,
      },
      Object {
        "active": false,
        "href": "/dashboard/maintab/?tab=compute",
        "iconClass": "pficon pficon-cpu",
        "id": "compute",
        "items": Array [
          Object {
            "active": false,
            "href": "/dashboard/maintab/?tab=clo",
            "iconClass": "fa fa-plus",
            "id": "clo",
            "items": Array [
              Object {
                "active": true,
                "href": "/ems_cloud/show_list",
                "iconClass": null,
                "id": "ems_cloud",
                "items": Array [],
                "title": "Providers",
                "type": "new_window",
                "visible": true,
              },
              Object {
                "active": false,
                "href": "/availability_zone/show_list",
                "iconClass": null,
                "id": "availability_zone",
                "items": Array [],
                "title": "Availability Zones",
                "type": "default",
                "visible": true,
              },
              Object {
                "active": false,
                "href": "/host_aggregate/show_list",
                "iconClass": null,
                "id": "host_aggregate",
                "items": Array [],
                "title": "Host Aggregates",
                "type": "new_window",
                "visible": false,
              },
            ],
            "title": "Clouds",
            "type": "new_window",
            "visible": true,
          },
        ],
        "title": "Compute",
        "type": "new_window",
        "visible": true,
      },
    ]
  }
>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <div
      className="container"
    >
      <div
        className="nav-pf-vertical nav-pf-vertical-with-sub-menus nav-pf-vertical-collapsible-menus"
        id="main-menu"
        onMouseLeave={[Function]}
      >
        <ul
          className="list-group"
          id="maintab"
        >
          <MenuItem
            active={true}
            handleSetActiveIds={[Function]}
            href="/dashboard/maintab/?tab=vi"
            iconClass="fa fa-dashboard"
            id="vi"
            items={
              Array [
                Object {
                  "active": true,
                  "href": "/dashboard/show",
                  "iconClass": null,
                  "id": "dashboard",
                  "items": Array [],
                  "title": "Dashboard",
                  "type": "modal",
                  "visible": true,
                },
                Object {
                  "active": false,
                  "href": "/chargeback/explorer",
                  "iconClass": null,
                  "id": "chargeback",
                  "items": Array [],
                  "title": "Chargeback",
                  "type": "big_iframe",
                  "visible": true,
                },
              ]
            }
            key="vi"
            level={0}
            title="Cloud Intel"
            type="default"
            visible={true}
          >
            <TopLevel
              active={true}
              handleSetActiveIds={[Function]}
              href="/dashboard/maintab/?tab=vi"
              iconClass="fa fa-dashboard"
              id="vi"
              items={
                Array [
                  Object {
                    "active": true,
                    "href": "/dashboard/show",
                    "iconClass": null,
                    "id": "dashboard",
                    "items": Array [],
                    "title": "Dashboard",
                    "type": "modal",
                    "visible": true,
                  },
                  Object {
                    "active": false,
                    "href": "/chargeback/explorer",
                    "iconClass": null,
                    "id": "chargeback",
                    "items": Array [],
                    "title": "Chargeback",
                    "type": "big_iframe",
                    "visible": true,
                  },
                ]
              }
              level={0}
              title="Cloud Intel"
              type="default"
              visible={true}
            >
              <li
                className="menu-list-group-item secondary-nav-item-pf active"
                id="menu_section_vi"
                onBlur={[Function]}
                onMouseEnter={[Function]}
              >
                <a
                  className="top-level-item"
                  href="/dashboard/maintab/?tab=vi"
                  onClick={[Function]}
                  target="_self"
                >
                  <span />
                  <span
                    className="list-group-item-value"
                  >
                    Cloud Intel
                  </span>
                </a>
                <div
                  className="nav-pf-secondary-nav"
                  id="menu-vi"
                >
                  <div
                    className="nav-item-pf-header"
                  >
                    <a
                      className="top-level-item"
                    >
                      <span>
                        Cloud Intel
                      </span>
                    </a>
                  </div>
                  <ul
                    className="list-group"
                  >
                    <MenuItem
                      active={true}
                      handleSetActiveIds={[Function]}
                      href="/dashboard/show"
                      iconClass={null}
                      id="dashboard"
                      items={Array []}
                      key="dashboard"
                      level={1}
                      title="Dashboard"
                      type="modal"
                      visible={true}
                    >
                      <SecondLevel
                        active={true}
                        handleSetActiveIds={[Function]}
                        href="/dashboard/show"
                        iconClass={null}
                        id="dashboard"
                        items={Array []}
                        level={1}
                        title="Dashboard"
                        type="modal"
                        visible={true}
                      >
                        <li
                          className="menu-list-group-item active"
                          id="menu_item_dashboard"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <a
                            onClick={[Function]}
                            target="_self"
                          >
                            <span
                              className="list-group-item-value"
                            >
                              Dashboard
                            </span>
                          </a>
                          <div
                            className="nav-pf-tertiary-nav"
                          >
                            <div
                              className="nav-item-pf-header"
                            >
                              <a>
                                <span>
                                  Dashboard
                                </span>
                              </a>
                            </div>
                          </div>
                        </li>
                      </SecondLevel>
                    </MenuItem>
                    <MenuItem
                      active={false}
                      handleSetActiveIds={[Function]}
                      href="/chargeback/explorer"
                      iconClass={null}
                      id="chargeback"
                      items={Array []}
                      key="chargeback"
                      level={1}
                      title="Chargeback"
                      type="big_iframe"
                      visible={true}
                    >
                      <SecondLevel
                        active={false}
                        handleSetActiveIds={[Function]}
                        href="/chargeback/explorer"
                        iconClass={null}
                        id="chargeback"
                        items={Array []}
                        level={1}
                        title="Chargeback"
                        type="big_iframe"
                        visible={true}
                      >
                        <li
                          className="menu-list-group-item"
                          id="menu_item_chargeback"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <a
                            href="/dashboard/iframe?id=chargeback"
                            onClick={[Function]}
                            target="_self"
                          >
                            <span
                              className="list-group-item-value"
                            >
                              Chargeback
                            </span>
                          </a>
                          <div
                            className="nav-pf-tertiary-nav"
                          >
                            <div
                              className="nav-item-pf-header"
                            >
                              <a>
                                <span>
                                  Chargeback
                                </span>
                              </a>
                            </div>
                          </div>
                        </li>
                      </SecondLevel>
                    </MenuItem>
                  </ul>
                </div>
              </li>
            </TopLevel>
          </MenuItem>
          <MenuItem
            active={false}
            handleSetActiveIds={[Function]}
            href="/dashboard/maintab/?tab=compute"
            iconClass="pficon pficon-cpu"
            id="compute"
            items={
              Array [
                Object {
                  "active": false,
                  "href": "/dashboard/maintab/?tab=clo",
                  "iconClass": "fa fa-plus",
                  "id": "clo",
                  "items": Array [
                    Object {
                      "active": true,
                      "href": "/ems_cloud/show_list",
                      "iconClass": null,
                      "id": "ems_cloud",
                      "items": Array [],
                      "title": "Providers",
                      "type": "new_window",
                      "visible": true,
                    },
                    Object {
                      "active": false,
                      "href": "/availability_zone/show_list",
                      "iconClass": null,
                      "id": "availability_zone",
                      "items": Array [],
                      "title": "Availability Zones",
                      "type": "default",
                      "visible": true,
                    },
                    Object {
                      "active": false,
                      "href": "/host_aggregate/show_list",
                      "iconClass": null,
                      "id": "host_aggregate",
                      "items": Array [],
                      "title": "Host Aggregates",
                      "type": "new_window",
                      "visible": false,
                    },
                  ],
                  "title": "Clouds",
                  "type": "new_window",
                  "visible": true,
                },
              ]
            }
            key="compute"
            level={0}
            title="Compute"
            type="new_window"
            visible={true}
          >
            <TopLevel
              active={false}
              handleSetActiveIds={[Function]}
              href="/dashboard/maintab/?tab=compute"
              iconClass="pficon pficon-cpu"
              id="compute"
              items={
                Array [
                  Object {
                    "active": false,
                    "href": "/dashboard/maintab/?tab=clo",
                    "iconClass": "fa fa-plus",
                    "id": "clo",
                    "items": Array [
                      Object {
                        "active": true,
                        "href": "/ems_cloud/show_list",
                        "iconClass": null,
                        "id": "ems_cloud",
                        "items": Array [],
                        "title": "Providers",
                        "type": "new_window",
                        "visible": true,
                      },
                      Object {
                        "active": false,
                        "href": "/availability_zone/show_list",
                        "iconClass": null,
                        "id": "availability_zone",
                        "items": Array [],
                        "title": "Availability Zones",
                        "type": "default",
                        "visible": true,
                      },
                      Object {
                        "active": false,
                        "href": "/host_aggregate/show_list",
                        "iconClass": null,
                        "id": "host_aggregate",
                        "items": Array [],
                        "title": "Host Aggregates",
                        "type": "new_window",
                        "visible": false,
                      },
                    ],
                    "title": "Clouds",
                    "type": "new_window",
                    "visible": true,
                  },
                ]
              }
              level={0}
              title="Compute"
              type="new_window"
              visible={true}
            >
              <li
                className="menu-list-group-item secondary-nav-item-pf"
                id="menu_section_compute"
                onBlur={[Function]}
                onMouseEnter={[Function]}
              >
                <a
                  className="top-level-item"
                  href="/dashboard/maintab/?tab=compute"
                  onClick={[Function]}
                  target="_blank"
                >
                  <span />
                  <span
                    className="list-group-item-value"
                  >
                    Compute
                  </span>
                </a>
                <div
                  className="nav-pf-secondary-nav"
                  id="menu-compute"
                >
                  <div
                    className="nav-item-pf-header"
                  >
                    <a
                      className="top-level-item"
                    >
                      <span>
                        Compute
                      </span>
                    </a>
                  </div>
                  <ul
                    className="list-group"
                  >
                    <MenuItem
                      active={false}
                      handleSetActiveIds={[Function]}
                      href="/dashboard/maintab/?tab=clo"
                      iconClass="fa fa-plus"
                      id="clo"
                      items={
                        Array [
                          Object {
                            "active": true,
                            "href": "/ems_cloud/show_list",
                            "iconClass": null,
                            "id": "ems_cloud",
                            "items": Array [],
                            "title": "Providers",
                            "type": "new_window",
                            "visible": true,
                          },
                          Object {
                            "active": false,
                            "href": "/availability_zone/show_list",
                            "iconClass": null,
                            "id": "availability_zone",
                            "items": Array [],
                            "title": "Availability Zones",
                            "type": "default",
                            "visible": true,
                          },
                          Object {
                            "active": false,
                            "href": "/host_aggregate/show_list",
                            "iconClass": null,
                            "id": "host_aggregate",
                            "items": Array [],
                            "title": "Host Aggregates",
                            "type": "new_window",
                            "visible": false,
                          },
                        ]
                      }
                      key="clo"
                      level={1}
                      title="Clouds"
                      type="new_window"
                      visible={true}
                    >
                      <SecondLevel
                        active={false}
                        handleSetActiveIds={[Function]}
                        href="/dashboard/maintab/?tab=clo"
                        iconClass="fa fa-plus"
                        id="clo"
                        items={
                          Array [
                            Object {
                              "active": true,
                              "href": "/ems_cloud/show_list",
                              "iconClass": null,
                              "id": "ems_cloud",
                              "items": Array [],
                              "title": "Providers",
                              "type": "new_window",
                              "visible": true,
                            },
                            Object {
                              "active": false,
                              "href": "/availability_zone/show_list",
                              "iconClass": null,
                              "id": "availability_zone",
                              "items": Array [],
                              "title": "Availability Zones",
                              "type": "default",
                              "visible": true,
                            },
                            Object {
                              "active": false,
                              "href": "/host_aggregate/show_list",
                              "iconClass": null,
                              "id": "host_aggregate",
                              "items": Array [],
                              "title": "Host Aggregates",
                              "type": "new_window",
                              "visible": false,
                            },
                          ]
                        }
                        level={1}
                        title="Clouds"
                        type="new_window"
                        visible={true}
                      >
                        <li
                          className="menu-list-group-item tertiary-nav-item-pf"
                          id="menu_section_clo"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <a
                            href="/dashboard/maintab/?tab=clo"
                            onClick={[Function]}
                            target="_blank"
                          >
                            <span
                              className="list-group-item-value"
                            >
                              Clouds
                            </span>
                          </a>
                          <div
                            className="nav-pf-tertiary-nav"
                          >
                            <div
                              className="nav-item-pf-header"
                            >
                              <a>
                                <span>
                                  Clouds
                                </span>
                              </a>
                            </div>
                            <ul
                              className="list-group"
                            >
                              <MenuItem
                                active={true}
                                href="/ems_cloud/show_list"
                                iconClass={null}
                                id="ems_cloud"
                                items={Array []}
                                key="ems_cloud"
                                level={2}
                                title="Providers"
                                type="new_window"
                                visible={true}
                              >
                                <ThirdLevel
                                  active={true}
                                  href="/ems_cloud/show_list"
                                  iconClass={null}
                                  id="ems_cloud"
                                  items={Array []}
                                  level={2}
                                  title="Providers"
                                  type="new_window"
                                  visible={true}
                                >
                                  <li
                                    className="menu-list-group-item active"
                                    id="menu_item_ems_cloud"
                                  >
                                    <a
                                      href="/ems_cloud/show_list"
                                      onClick={[Function]}
                                      target="_blank"
                                    >
                                      <span
                                        className="list-group-item-value"
                                      >
                                        Providers
                                      </span>
                                    </a>
                                  </li>
                                </ThirdLevel>
                              </MenuItem>
                              <MenuItem
                                active={false}
                                href="/availability_zone/show_list"
                                iconClass={null}
                                id="availability_zone"
                                items={Array []}
                                key="availability_zone"
                                level={2}
                                title="Availability Zones"
                                type="default"
                                visible={true}
                              >
                                <ThirdLevel
                                  active={false}
                                  href="/availability_zone/show_list"
                                  iconClass={null}
                                  id="availability_zone"
                                  items={Array []}
                                  level={2}
                                  title="Availability Zones"
                                  type="default"
                                  visible={true}
                                >
                                  <li
                                    className="menu-list-group-item"
                                    id="menu_item_availability_zone"
                                  >
                                    <a
                                      href="/availability_zone/show_list"
                                      onClick={[Function]}
                                      target="_self"
                                    >
                                      <span
                                        className="list-group-item-value"
                                      >
                                        Availability Zones
                                      </span>
                                    </a>
                                  </li>
                                </ThirdLevel>
                              </MenuItem>
                              <MenuItem
                                active={false}
                                href="/host_aggregate/show_list"
                                iconClass={null}
                                id="host_aggregate"
                                items={Array []}
                                key="host_aggregate"
                                level={2}
                                title="Host Aggregates"
                                type="new_window"
                                visible={false}
                              >
                                <ThirdLevel
                                  active={false}
                                  href="/host_aggregate/show_list"
                                  iconClass={null}
                                  id="host_aggregate"
                                  items={Array []}
                                  level={2}
                                  title="Host Aggregates"
                                  type="new_window"
                                  visible={false}
                                />
                              </MenuItem>
                            </ul>
                          </div>
                        </li>
                      </SecondLevel>
                    </MenuItem>
                  </ul>
                </div>
              </li>
            </TopLevel>
          </MenuItem>
        </ul>
      </div>
    </div>
  </Grid>
</MainMenu>
`;
